src/commands.rs

```diff
@@ -3734,6 +3702,9 @@ fn cmd_workspace_list(
     for (workspace_id, checkout_id) in repo.view().wc_commit_ids().iter().sorted() {
         write!(ui, "{}: ", workspace_id.as_str())?;
         let commit = repo.store().get_commit(checkout_id)?;
+        // TODO: Should the "current" workspace be the one user operates on or the one
+        // currently listing? Here workspace_id != workspace_command.workspace_id(), and
+        // that's why workspace_command.write_commit_summary() isn't used.
```

Comment:
Good question. I'd say it should be the current workspace, so the current behavior is incorrect. What do you think?

Reply:
Yeah, it's more consistent with the other commands, and simpler. I'll fix that.